include/rx/ranges.hpp

```diff
@@ -2317,6 +2317,109 @@ template <class... Inputs>
     return ZipLongestRange(std::forward_as_tuple(as_input_range(std::forward<Inputs>(inputs))...));
 }
 
+/// Forward all elements of the range as-is.
+struct passthrough {
+    template <class R>
+    struct Range {
+        using output_type = get_output_type_of_t<R>;
+        static constexpr bool is_finite = is_finite_v<R>;
+        static constexpr bool is_idempotent = is_idempotent_v<R>;
+
+        R input;
+
+        constexpr void next() {
+            RX_ASSERT(!at_end());
+            input.next();
+        }
+
+        [[nodiscard]] constexpr decltype(auto) get() const {
+            RX_ASSERT(!at_end());
+            return input.get();
+        }
+
+        [[nodiscard]] constexpr bool at_end() const {
+            return input.at_end();
+        }
+
+        [[nodiscard]] constexpr size_t size_hint() const {
+            return input.size_hint();
+        }
+    };
+
+    template <class InputRange>
+    [[nodiscard]] constexpr auto operator()(InputRange&& input) const {
+        using Inner = get_range_type_t<InputRange>;
+        return Range<Inner>{as_input_range(std::forward<InputRange>(input))};
+    }
+};
+
+/// Present elements in a range to a function, then pass the values along as-is.
+template <class F>
+struct spy {
+    F func;
+
+    template <class Fx>
+    explicit constexpr spy(Fx&& func) : func(std::forward<Fx>(func)) {}
```

Comment:
Taking `F` by value would be less problematic.

Reply:
Maybe this instead?
```diff
- template <class F> spy(F&&)->spy<remove_cvref_t<F>>;
+ template <class F> spy(F)->spy<remove_cvref_t<F>>;
```

Reply:
yeah it prevents the copy/move constructor to be called

Reply:
So I think the deduction guide is fine, just the constructor might as well take `F` by value. :-)